sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/KalixServer.scala

```diff
@@ -403,7 +405,29 @@ case class KalixServer(applicationContext: ApplicationContext, config: Config) {
 
         setKalixClient(clz, webClientProviderHolder)
 
-        kalixBeanFactory.getBean(clz)
+        val workflowEntity = kalixBeanFactory.getBean(clz)
+
+        val workflowStateType: Class[S] =
+          workflowEntity.getClass.getGenericSuperclass
+            .asInstanceOf[ParameterizedType]
+            .getActualTypeArguments
+            .head
+            .asInstanceOf[Class[S]]
+
+        messageCodec.lookupTypeHint(workflowStateType)
```

Comment:
we need to register state type as well, for commands that was handled with by `_extractAndSetCurrentState` method, but this method is not invoked for a step invocation